Geometry/HGCalCommonData/src/HGCalWaferType.cc

```diff
@@ -70,3 +106,31 @@ std::pair<double,double> HGCalWaferType::rLimits(double zpos) {
   return std::pair<double,double>(rfine*HGCalParameters::k_ScaleToDDD,rcoarse*HGCalParameters::k_ScaleToDDD);
 }
 
+double HGCalWaferType::areaPolygon(std::vector<double> const& x,
+				   std::vector<double> const& y) {
+  double area = 0.0;
+  for (unsigned int k1=0; k1<x.size(); ++k1) {
+    unsigned int k2 = (k1 == x.size()-1) ? 0 : k1+1;
+    area += 0.5*(x[k1]*y[k2]-x[k2]*y[k1]);
+  }
+  return area;
+}
+
+std::pair<double,double> HGCalWaferType::intersection(int k1, int k2, 
```

Comment:
@bsunanda - what do you try to calculate here?

Reply:
Intersection of a circle of radius rr (origin 0,0) and straight line going through points k1 and k2. Out of the 2 solutions choose the one closest to xpos,typos